packages/gatsby-link/src/index.js

```diff
@@ -178,6 +178,11 @@ GatsbyLink.propTypes = {
   replace: PropTypes.bool,
 }
 
+const showWarningMessage = action =>
```

Comment:
Want to call the function showDeprecationWarning and accept functionName, altFunctionName and version?

Reply:
Would've preferred to take `altFunctionName` and `version` too as parameters too (so we could reuse this function) but this is valuable as well so thank you!